probe/ebpf/flow.c

```diff
@@ -98,14 +98,23 @@ static inline void fill_transport(struct __sk_buff *skb, __u8 protocol, int offs
 	update_hash_half(&hash_dst, layer->port_dst);
 
 	switch (protocol) {
-		case IPPROTO_UDP:
-			//len -= sizeof(struct udphdr);
-			fill_payload(skb, offset, flow, len);
-			break;
-		case IPPROTO_TCP:
-			//len -= sizeof(struct tcphdr);
-			fill_payload(skb, offset, flow, len);
-			break;
+        case IPPROTO_UDP:
+            offset+=sizeof(struct udphdr);
+            len -= sizeof(struct udphdr);
+            fill_payload(skb, offset, flow, len);
+            break;
+        case IPPROTO_TCP:
+            __u8 flags = load_byte(skb,offset+14);
+            layer->ab_syn=(flags&0x02) > 0 ? tm : 0;
+            layer->ab_fin=(flags&0x01) > 0 ? tm : 0;
+            layer->ab_rst=(flags&0x04) > 0 ? tm : 0;
+            layer->ba_syn=0;
```

Comment:
the following three line are useless as the layer as been initialized to 0

Reply:
done